exasol/nb_connector/itde_manager.py

```diff
@@ -85,18 +85,37 @@ def bring_itde_up(conf: Secrets) -> None:
     conf.save(AILabConfig.cert_vld, "False")
 
 
-def _add_current_container_to_db_network(container_info: ContainerInfo):
-    network_name = container_info.network_info.network_name
+def _get_current_container(docker_client: docker.DockerClient):
+    ip_addresses = _get_ipv4_ddresses()
+    return ContainerByIp(docker_client).find(ip_addresses)
+
+
+def _add_current_container_to_db_network(network_name: str) -> None:
     with ContextDockerClient() as docker_client:
-        ip_addresses = _get_ipv4_ddresses()
-        container = ContainerByIp(docker_client).find(ip_addresses)
+        container = _get_current_container(docker_client)
         if not container:
             return
         network = _get_docker_network(docker_client, network_name)
-        if network:
+        if network and (container not in network.containers):
             network.connect(container.id)
 
 
+def _is_current_container_not_in_db_network(network_name: str) -> bool:
```

Comment:
How does this function relate to `self._add_current_container_to_db_network()`? 
I have the feeling that part of `self._add_current_container_to_db_network()` can be replaced by using this function?